modules/gapi/include/opencv2/gapi/gmat.hpp

```diff
@@ -46,6 +46,19 @@ class GAPI_EXPORTS GMat
     std::shared_ptr<GOrigin> m_priv;
 };
 
+class GAPI_EXPORTS GMatP
+{
+public:
+    GMatP();                                 // Empty constructor
+    GMatP(const GNode &n, std::size_t out);  // Operation result constructor
+
+    GOrigin& priv();                        // Internal use only
+    const GOrigin& priv()  const;           // Internal use only
+
+private:
+    std::shared_ptr<GOrigin> m_priv;
+};
```

Comment:
Could this be just subclassed from regular `cv::GMat` (with all fields reused)}

Reply:
Yes, will fix